tests/test_fallback_streams.py

```
@@ -2283,25 +2293,32 @@ def test_selection_fallback_does_not_wait_for_optional_tail_after_max_filled(
+            # record whether the scan waited for it -- load-independent. The 2s
+            # cap only bounds a regression where the scan blocks on this fetch.
+            release_slow.wait(timeout=2)
+            tail_completed[0] = True
         return manifests[url]
 
     mock_fetch.side_effect = fetch
-    release_timer = threading.Timer(0.25, release_slow.set)
-    release_timer.start()
     try:
-        before = _time.monotonic()
         attach_fallback_candidates_for_selection(selected, [selected] + candidates)
```

Ruff (0.15.18)

[warning] Consider `[selected, *candidates]` instead of concatenation

Replace with `[selected, *candidates]`

(RUF005)